statefulset/statefulset.go

```diff
@@ -0,0 +1,70 @@
+package statefulset
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/vshn/statefulset-resize-controller/pvc"
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+// FailedLabel is a label for failed sts resizing that need human interaction
+const FailedLabel = "sts-resize.appuio.ch/failed"
+
+// PvcAnnotation is an annotation in which the initial state of the pvcs is stored in
+const PvcAnnotation = "sts-resize.appuio.ch/pvcs"
+
+// Entity contains all data to manage a statfulset resizing
+type Entity struct {
+	Old  *appsv1.StatefulSet
+	Pvcs []pvc.Entity
+
+	sts *appsv1.StatefulSet
+}
+
+// NewEntity return a new StatefulSet Info
+func NewEntity(sts *appsv1.StatefulSet) (*Entity, error) {
+	si := Entity{}
+	si.sts = sts.DeepCopy()
+	si.Old = sts
+	if sts.Annotations[PvcAnnotation] != "" {
+		if err := json.Unmarshal([]byte(sts.Annotations[PvcAnnotation]), &si.Pvcs); err != nil {
+			return nil, fmt.Errorf("Annotation %s malformed", PvcAnnotation)
+		}
+	}
+	return &si, nil
+}
+
+// Sts returns the updated StatefulSet resource
+func (s *Entity) Sts() (*appsv1.StatefulSet, error) {
```

Comment:
Please don't use abbreviations for function names.
```suggestion
// Sts returns the updated StatefulSet resource
func (s *Entity) StatefulSet() (*appsv1.StatefulSet, error) {
```